src/test/components/AppLocalizationProvider.test.js

```diff
@@ -9,51 +9,75 @@ import { AppLocalizationProvider } from 'firefox-profiler/components/app/AppLoca
 import { blankStore } from '../fixtures/stores';
 import { Provider } from 'react-redux';
 import {
-  getIsL10nFetching,
+  getl10nFetchingPhase,
   getLocalization,
 } from 'firefox-profiler/selectors/l10n';
 import { lazilyParsedBundles } from 'firefox-profiler/app-logic/l10n';
 import { requestL10n, receiveL10n } from 'firefox-profiler/actions/l10n';
-import { ReactLocalization } from '@fluent/react';
+import { ReactLocalization, Localized } from '@fluent/react';
+import { coerceMatchingShape } from '../../utils/flow';
+import { waitUntilState } from '../fixtures/utils';
 
 describe('AppLocalizationProvider', () => {
-  beforeEach(function() {
-    const response = (({
-      ok: true,
-      status: 200,
-      headers: {
-        get: () => 'text/plain',
-      },
-      text: () => Promise.resolve('resolved'),
-    }: any): Response);
-    window.fetch = jest.fn().mockResolvedValue(response);
-  });
+  function setupL10nFetch(language) {
+    jest.spyOn(window.navigator, 'languages', 'get').mockReturnValue(['en-US']);
+
+    const fetch = jest.fn().mockImplementation((fetchUrl: string) => {
+      if (fetchUrl === `/locales/${language}/app.ftl`) {
+        const response = (({
+          ok: true,
+          status: 200,
+          headers: {
+            get: () => 'text/plain',
+          },
+          text: () => Promise.resolve(`test-id = This is ${language} Text`),
+        }: any): Response);
+
+        return Promise.resolve(response);
+      }
+      return Promise.reject(
+        coerceMatchingShape<Response>({
+          ok: false,
+          status: 404,
+          statusText: 'Not found',
+        })
+      );
+    });
+    (window: any).fetch = fetch;
+  }
 
   afterEach(function() {
     delete window.fetch;
   });
 
-  it('changes the state accordingly when the actions are dispatched', () => {
+  function setup(language = 'en-US') {
     const store = blankStore();
     const { dispatch, getState } = store;
+    const translatedText = `This is ${language} Text`;
```

Comment:
Hmm, we don't use this variable at all in the fetch mock. My suggestion was to return this in the fetch mock also, so we can have the assertion. Right now this won't work if we put another language.
I think we can merge the `setupL10nFetch` and `setup` together and just keep one single `setup` instead because we are not using the setupL10nFetch function anywhere else.